src/events/Windows.cpp

```diff
@@ -767,11 +767,23 @@ void Events::listener_fullscreenWindow(void* owner, void* data) {
         if (REQUESTED->fullscreen && PWINDOW->m_bIsFullscreen) {
             const auto PWORKSPACE = g_pCompositor->getWorkspaceByID(PWINDOW->m_iWorkspaceID);
             if (PWORKSPACE->m_efFullscreenMode != FULLSCREEN_FULL) {
+                // Store that we were maximized
+                PWINDOW->m_bWasMaximized = true;
```

Comment:
why not plop setting of thise into CCompositor::setWindowFullscreen?

Reply:
I'm pretty sure, the only way to trigger a window to switch from maximize -> fullscreen is by the application itself. Trying to fullscreen with e.g. a hyprland keybind just switches back to normal. 
Besides that, then I should probably also put the switching code (from and to fullscreen/maximize) into setWindowFullscreen(Then manual maximize -> fullscreen would work the same way). The problem here is, that many places in the codebase assume, that you cannot switch from one fullscreen mode to the other(This is bypassed here). I'd have to disable all these checks for things to work correctly again (I actually did that before, code is in fullscreen-fixes branch of my fork), which isn't ideal.

Reply:
ah yeah right. forgor :skull: